### ISSUE TITLE
[BUG] "MSBUILD : error MSB1008: Only one project can be specified" caused by double quotes

### ISSUE THREAD

**Issue by baebranch** [microsoft/vscode-dotnettools#1578]
### Describe the Issue

This seems to be an issue with the build command itself. This is also preventing the debugger from starting when debugging for android platform. Windows local machine debugging works fine.

For this build command:
`Executing task: dotnet build -t:Build -p:Configuration=Debug -f net8.0-android -p:AndroidSdkDirectory="C:\Users\<user>\AppData\Local\Android\sdk" "-p:JavaSdkDirectory="C:\Program Files\Java\jdk-17"" C:\code\csharp\hello_world\MauiApp1\MauiApp1.csproj`

This section:
`"-p:JavaSdkDirectory="C:\Program Files\Java\jdk-17""`

should be replaced by this:
`-p:JavaSdkDirectory="C:\Program Files\Java\jdk-17"`

### Steps To Reproduce

1. Install .NET MAUI extension
2. Select ".NET: New Project" in vscode command palette
3. Configure target debug device to "Android (build only)" by clicking "{}" in bottom status bar
3. "F5" to start debugging
4. Pop up shows "Error processing launch options at field: program"

### Expected Behavior

The app should start debugging on the android emulator

### Environment Information

- .NET MAUI vscode extension version: `v1.5.19 (pre-release)`
- C# Dev Kit vscode extension version: `v1.12.37 (pre-release)`
- C# vscode extension version: `v2.50.27`
- dotnet --version `8.0.204`
- dotnet workload list output
```
Installed Workload Id      Manifest Version      Installation Source
--------------------------------------------------------------------
android                     34.0.143/8.0.100       SDK 8.0.200
maui                        8.0.82/8.0.100         SDK 8.0.200
maui-android                8.0.82/8.0.100         SDK 8.0.200
```

- .Net Maui vscode Output
```
ANDROID ENVIRONMENT STATUS: 11/2/2024, 11:36:29 AM
 
Here is the status of your Android development environment:
 
Required components:
 
	✓ Java SDK - INSTALLED (version '17.0.6')
		- Path: 'C:\Program Files\Java\jdk-17' (Workload)
		- Recommended detected locations:
			- 'c:\program files\common files\oracle\java' (PATH)

	✓ Android SDK - INSTALLED
		- Path: 'C:\Users\<user>\AppData\Local\Android\sdk' (VSCode settings)
 
		✓ platforms/android-34 - INSTALLED (version '3')
		✓ build-tools/34.0.0 - INSTALLED (version '34.0.0')
		✓ platform-tools - INSTALLED (version '35.0.2')
		✓ cmdline-tools/11.0 - INSTALLED (version '11.0')
 
Optional components:
 
	✓ Android Virtual Device (AVD) - COMPLETE
		- AVD Path: 'C:\Users\<user>\.android\avd'
		- AVD devices - DETECTED ('Pixel8API35')
 
		✓ emulator - INSTALLED (version '35.2.10')
		✓ system-images/android-34/google_apis/x86_64 - INSTALLED (version '14')
		✓ system-images/android-35/google_apis_playstore/x86_64 - INSTALLED (version '8')
 
IMPORTANT:
- If you are sharing the Android SDK location with Android Studio, the recommended way to install/update the SDK and accept licenses is by using Android Studio.
```

**Comment by mauroa** [comment]
Hi @baebranch, according to the repro steps you provide, you shouldn't expect the app to start debug if you are selecting an "Android (build only)". The purpose of that build only debug target is to just build your project, because for now the "right click > Build solution" option is not supported for MAUI.
Could you please try to select a real Android emulator or device and try to debug again, seeing if that makes any change in the repro?
Thanks

**Comment by baebranch** [comment]
> Hi [@baebranch](https://github.com/baebranch), according to the repro steps you provide, you shouldn't expect the app to start debug if you are selecting an "Android (build only)". The purpose of that build only debug target is to just build your project, because for now the "right click > Build solution" option is not supported for MAUI. Could you please try to select a real Android emulator or device and try to debug again, seeing if that makes any change in the repro? Thanks

Sorry I came across a few bugs and didn't update the issue properly. That was related to a pre-launch command that was doubled quoted and causing errors, which I replaced myself. However I've selected an android emulator device, I was just indicating that in the build only option it built with my updated prelaunch build task, loads onto the emulated device and then then shows that error popup.

**Comment by baebranch** [comment]
> Thanks [@baebranch](https://github.com/baebranch).
> 
> Regarding this: "That was related to a pre-launch command that was doubled quoted and causing errors, which I replaced myself.", I'm not clear if you got unblocked or if you are still having issues. In case you still have issues, could you share the launch.json with the custom steps that you are using?

I upgraded to .net 9 and the app loads correctly on the emulator but gets stuck at "waiting for debugger" message or just stuck on the splash screen. There are also issues recognizing the running AVD and discovering all of the configured AVDs.

This is the modified task in `tasks.json`
```
{
	"version": "2.0.0",
	"tasks": [
		{
			"label": "Build",
			"type": "shell",
			"command": "C:\\WINDOWS\\System32\\cmd.exe",
			"args": [
				"/d",
				"/c",
				"dotnet",
				"build",
				"-t:Run",
				"-p:Configuration=Debug",
				"-f",
				"net9.0-android",
				"-p:AdbTarget=-s%20emulator-5554",
				"-p:AndroidAttachDebugger=true",
				"-p:AndroidSdbTargetPort=53027",
				"-p:AndroidSdbHostPort=53027",
				"-p:CustomAfterMicrosoftCSharpTargets=\"c:\\Users\\<user>\\.vscode\\extensions\\ms-dotnettools.dotnet-maui-1.6.4-win32-x64\\dist\\resources\\Custom.After.Microsoft.CSharp.targets\"",
				"-p:MauiVSCodeBuildOutputFile=\"C:\\Users\\<user>\\AppData\\Local\\Temp\\dotnet-maui\\maui-vsc-36c1a1e8-5bd3-402f-a64d-8d8cc8307f89.json\"",
				"-p:AndroidSdkDirectory=\"C:\\Users\\<user>\\AppData\\Local\\Android\\sdk\"",
				"-p:JavaSdkDirectory=\"C:\\Program Files (x86)\\Android\\openjdk\\jdk-17.0.12\"",
				"C:\\code\\csharp\\new_test\\MauiApp1\\MauiApp1.csproj"
			],
			"problemMatcher": [
				"$msCompile",
				"maui-problem-matcher-location",
				"maui-problem-matcher",
				"maui-problem-matcher-location-multiline",
				"maui-problem-matcher-multiline"
			],
		}
	]
}
```

**Comment by mauroa** [comment]
Hi @baebranch, sorry for the delay on this.

We discourage the usage you are doing with the tasks.json, since you are basically trying to do what the .NET MAUI extension should do for you, and this could cause unwanted behaviors, especially if you are using `-t:Run` in it. The idea is that our extensions can provide you with all you need to make your scenario work. If something doesn't work as you expect, we should go over it and try to make it work without workarounds.

I would like to first know the reason and need for using this tasks.json instead of letting VS Code and .NET MAUI do it for you. Was it a particular reason or were you just trying to find the way to make it work?

I would also suggest trying to remove it and compare the behavior without it. To do so, could you please follow the steps below?

1 - Remove the custom task in the tasks.json
2 - Update C#, C# DevKit and .NET MAUI to latest pre-release versions (this will bring important fixes for the debugger, for the project and solution management, launch experience, etc.)
3 - Make sure the following VS Code setting is enabled:

![Image](https://github.com/user-attachments/assets/583aef92-ef5a-4ef5-a441-9816d9925900)

4 - Try selecting again the startup project and launch configuration (debug target) using the "{}"

5 - Hit F5, select C# from the debuggers list, and let VS code build and launch your application in the selected debug target (make sure it's not build only, since that will not launch anything)

In case something goes wrong, please share the content of the Terminal output and also the Debug Console output, so we can see what's going on in there.

Thanks

**Comment by AllenD-MSFT** [comment]
We intend to close this issue next week since we need more info.

**Comment by baebranch** [comment]
My build output:

```
Executing task: dotnet build -t:Run -p:Configuration=Debug -f net9.0-android -p:AdbTarget=-s%20emulator-5554 -p:AndroidAttachDebugger=true -p:AndroidSdbTargetPort=58342 -p:AndroidSdbHostPort=58342 -p:CustomAfterMicrosoftCSharpTargets="c:\Users\<user>\.vscode\extensions\ms-dotnettools.dotnet-maui-1.7.10-win32-x64\dist\resources\Custom.After.Microsoft.CSharp.targets" -p:MauiVSCodeBuildOutputFile="C:\Users\<user>\AppData\Local\Temp\dotnet-maui\maui-vsc-1c75188b-f522-427a-987f-718a650aa4a0.json" -p:AndroidSdkDirectory="C:\Users\<user>\AppData\Local\Android\sdk" "-p:JavaSdkDirectory="C:\Program Files (x86)\Android\openjdk\jdk-17.0.12"" C:\code\csharp\demo\MauiApp1\MauiApp1.csproj 

MSBUILD : error MSB1008: Only one project can be specified.
    Full command line: 'C:\Program Files\dotnet\sdk\9.0.100\MSBuild.dll -maxcpucount -verbosity:m -tlp:default=auto -nologo -target:Restore -tlp:verbosity=quiet --property:Configuration=Debug --property:AdbTarget=-s%20emulator-5554 --property:AndroidAttachDebugger=true --property:AndroidSdbTargetPort=58342 --property:AndroidSdbHostPort=58342 --property:CustomAfterMicrosoftCSharpTargets=c:\Users\<user>\.vscode\extensions\ms-dotnettools.dotnet-maui-1.7.10-win32-x64\dist\resources\Custom.After.Microsoft.CSharp.targets --property:MauiVSCodeBuildOutputFile=C:\Users\<user>\AppData\Local\Temp\dotnet-maui\maui-vsc-1c75188b-f522-427a-987f-718a650aa4a0.json --property:AndroidSdkDirectory=C:\Users\<user>\AppData\Local\Android\sdk --property:JavaSdkDirectory=C:\Program Files (x86)\Android\openjdk\jdk-17.0.12 C:\code\csharp\demo\MauiApp1\MauiApp1.csproj -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\9.0.100\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\9.0.100\dotnet.dll'
  Switches appended by response files:
Switch: (x86)\Android\openjdk\jdk-17.0.12

For switch syntax, type "MSBuild -help"

 *  The terminal process "C:\WINDOWS\System32\cmd.exe /d /c dotnet build -t:Run -p:Configuration=Debug -f net9.0-android -p:AdbTarget=-s%20emulator-5554 -p:AndroidAttachDebugger=true -p:AndroidSdbTargetPort=58342 -p:AndroidSdbHostPort=58342 -p:CustomAfterMicrosoftCSharpTargets="c:\Users\<user>\.vscode\extensions\ms-dotnettools.dotnet-maui-1.7.10-win32-x64\dist\resources\Custom.After.Microsoft.CSharp.targets" -p:MauiVSCodeBuildOutputFile="C:\Users\<user>\AppData\Local\Temp\dotnet-maui\maui-vsc-1c75188b-f522-427a-987f-718a650aa4a0.json" -p:AndroidSdkDirectory="C:\Users\<user>\AppData\Local\Android\sdk" "-p:JavaSdkDirectory="C:\Program Files (x86)\Android\openjdk\jdk-17.0.12"" C:\code\csharp\demo\MauiApp1\MauiApp1.csproj" terminated with exit code: 1. 
```

**Comment by IoMarius** [comment]
I am also being troubled by this issue, but I do not use such a tasks.json file

**Comment by baebranch** [comment]
> Hi [@baebranch](https://github.com/baebranch), sorry for the delay on this.
> 
> We discourage the usage you are doing with the tasks.json, since you are basically trying to do what the .NET MAUI extension should do for you, and this could cause unwanted behaviors, especially if you are using `-t:Run` in it. The idea is that our extensions can provide you with all you need to make your scenario work. If something doesn't work as you expect, we should go over it and try to make it work without workarounds.
> 
> I would like to first know the reason and need for using this tasks.json instead of letting VS Code and .NET MAUI do it for you. Was it a particular reason or were you just trying to find the way to make it work?
> 
> I would also suggest trying to remove it and compare the behavior without it. To do so, could you please follow the steps below?
> 
> 1 - Remove the custom task in the tasks.json 2 - Update C#, C# DevKit and .NET MAUI to latest pre-release versions (this will bring important fixes for the debugger, for the project and solution management, launch experience, etc.) 3 - Make sure the following VS Code setting is enabled:
> 
> ![Image](https://github.com/user-attachments/assets/583aef92-ef5a-4ef5-a441-9816d9925900)
> 
> 4 - Try selecting again the startup project and launch configuration (debug target) using the "{}"
> 
> 5 - Hit F5, select C# from the debuggers list, and let VS code build and launch your application in the selected debug target (make sure it's not build only, since that will not launch anything)
> 
> In case something goes wrong, please share the content of the Terminal output and also the Debug Console output, so we can see what's going on in there.
> 
> Thanks

I've set up the environment on another computer from scratch and it worked. 

I however on my original device have also set the above mentioned settings and get the error "Task 'Build' not found" sometimes or the original error mentioned. Too many things can go wrong with debugger causing different errors.

**Comment by jamnic1994** [comment]
I am having the exact same issue when trying to avoid using a launch.json or a tasks.jason. This is the default run command that is put into the terminal when I click run and debug:

 *  Executing task: dotnet build -t:Run -p:Configuration=Debug -f net8.0-android -p:AdbTarget=-s%20emulator-5554 -p:AndroidAttachDebugger=true -p:AndroidSdbTargetPort=58872 -p:AndroidSdbHostPort=58872 -p:CustomAfterMicrosoftCSharpTargets="c:\Users\jamie\.vscode\extensions\ms-dotnettools.dotnet-maui-1.8.9-win32-x64\dist\resources\Custom.After.Microsoft.CSharp.targets" -p:MauiVSCodeBuildOutputFile="C:\Users\jamie\AppData\Local\Temp\dotnet-maui\maui-vsc-9ddd5c46-f655-4d71-85be-2549b7e966c1.json" -p:AndroidSdkDirectory="C:\Users\jamie\MAUI\android" "-p:JavaSdkDirectory="C:\Program Files\Eclipse Adoptium\jdk-17.0.14.7-hotspot"" -p:XamlTools="c:\Users\jamie\.vscode\extensions\ms-dotnettools.csharp-2.63.32-win32-x64\.xamlTools" C:\Users\jamie\Desktop\Uni\vault\Notes\Notes\Notes.csproj

With the error being:

- MSBUILD : error MSB1008: Only one project can be specified.

The issue I think is with this part of the default task command:

- "-p:JavaSdkDirectory="C:\Program Files\Eclipse Adoptium\jdk-17.0.14.7-hotspot""

There seems to be a set of "" marks that as far as I can tell shouldn't be there.

**Comment by Roneri75** [comment]
What is the status on this?
Is there any workaround for it?
I have tried evertyhing in here and still getting the same problem running on the latest prerelease versions.

**Comment by emaf** [comment]
Closing this as there are multiple issues reported here, and the last one seems to not be reproducible anymore. I can see the build arguments being correctly quoted when paths contain spaces, for instance I can see this using the latest Prerelease version when running from both Windows and macOS: `'-p:JavaSdkDirectory="C:\Program Files\Android\openjdk\jdk-21.0.8"'`.

If you are still facing any of this issues, please report it as a new issue and share the full build output including the executing task arguments.

**Comment by emaf** [comment]
I'm reopening this because I just realized the problem occurs when using `Command Prompt` as default terminal in VS Code. As a workoarund you can switch `Terminal › Integrated › Default Profile: Windows` to PowerShell or bash.

**Comment by Dev-PKT** [comment]
> I'm reopening this because I just realized the problem occurs when using `Command Prompt` as default terminal in VS Code. As a workoarund you can switch `Terminal › Integrated › Default Profile: Windows` to PowerShell or bash.

Here is the Genius. Worked for me.

**Comment by emaf** [comment]
This is now fixed in the .NET MAUI prerelease channel (v1.12.9).
